File: jianzhi_offer/cpp/03.cpp
class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
       set<int> seen;
       for (const auto& i: nums) {
           if (seen.find(i) != seen.end()) {
               return i;
           } else {
               seen.emplace(i);
           }
       }
       return -1;
    }
};

File: jianzhi_offer/py/03.py
class Solution(object):
    def findRepeatNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = sorted(nums)
        for i in range(len(nums)):
            if (nums[i] == nums[i+1]):
                return nums[i]